tensorflow_lite_support/cc/task/processor/image_preprocessor.cc

```diff
@@ -119,6 +145,29 @@ absl::Status ImagePreprocessor::Preprocess(const FrameBuffer& frame_buffer,
                            frame_buffer.dimension().height;
   }
 
+  // If dynamic, it will re-dim the entire graph as per the input.
+  if (is_input_dynamic_) {    
+    const TfLiteIntArray* dims = Tensor()->dims;
+    const Interpreter* interpreter = engine_->interpreter();
+
+    // The expected BxHxWxD values of the model.
+    const int expected_batch = Tensor()->dims->data[0];
+    const int expected_height = Tensor()->dims->data[1];
+    const int expected_width = Tensor()->dims->data[2];
+    const int expected_depth = Tensor()->dims->data[3];
+
+    // HxW of the input image.
+    const int image_height = roi.height();
+    const int image_width = roi.width();
+
+    const int height = height_mutable ? image_height : expected_height;
+    const int width = width_mutable ? image_width : expected_width;
```

Comment:
Just use input_specs_.image_width / input_specs_.image_height for width / height?

Again, no need to add const in front.

Reply:
That's ingenious, thanks!

Reply:
About the const, I'm doing it to establish that the values won't change. imo, it will improve readability and help the compiler as well.